src/azure-cli/azure/cli/command_modules/acr/_utils.py

```diff
@@ -469,3 +453,125 @@ def prepare_source_location(cmd, source_location, client_registries, registry_na
 class ResourceNotFound(CLIError):
     """For exceptions that a resource couldn't be found in user's subscription
     """
+
+
+# Scope & Tokens help functions
+def parse_repositories_from_actions(actions):
+    if not actions:
+        return []
+    from .scope_map import RepoScopeMapActions
+    valid_actions = {action.value for action in RepoScopeMapActions}
+    repositories = []
+    REPOSITORY = 'repositories/'
+    for action in actions:
+        if action.startswith(REPOSITORY):
+            for rule in valid_actions:
+                if action.endswith(rule):
+                    repo = action[len(REPOSITORY):-len(rule)-1]
+                    repositories.append(repo)
+    return list(set(repositories))
+
+def parse_actions_from_repositories(repository_actions_list):
+    if not repository_actions_list:
+        return []
+    from .scope_map import RepoScopeMapActions
+    valid_actions = {action.value for action in RepoScopeMapActions}
+    actions = []
+    for rule in repository_actions_list:
+        repository = rule[0]
+        if len(rule) < 2:
+            raise CLIError('At least one action must be specified with the repository {}.'.format(repository))
+        for action in rule[1:]:
+            action = action.lower()
+            if action not in valid_actions:
+                raise CLIError('Invalid action "{}" provided. \nValid actions are {}.'.format(action, valid_actions))
+            actions.append('{}/{}/{}'.format('repositories', repository, action))
+
+    return actions
+
+
+def parse_actions_from_gateways(gateway_actions_list):
```

Comment:
This code looks similar to parse_actions_from_repositories, wondering if this can be a shared function?

Reply:
we could but, it was just easier like this to handle different error messages. It didn't seemed worth the trouble. We can save a ifs as well.